WordPress/Classes/Networking/GravatarServiceRemote.swift

```diff
@@ -0,0 +1,112 @@
+        guard let targetURL = NSURL(string: UploadParameters.endpointURL) else {
+            assertionFailure()
+            return
+        }
+        
+        // Boundary
+        let boundary = boundaryForRequest()
+        
+        // Request
+        let request = NSMutableURLRequest(URL: targetURL)
+        request.HTTPMethod = UploadParameters.HTTPMethod
+        request.setValue("Bearer \(accountToken)", forHTTPHeaderField: "Authorization")
+        request.setValue("multipart/form-data; boundary=\(boundary)", forHTTPHeaderField: "Content-Type")
+        
+        // Body
+        let gravatarData = UIImageJPEGRepresentation(image, UploadParameters.compressionRatio)!
+        let requestBody = bodyWithGravatarData(gravatarData, account: accountEmail, boundary: boundary)
+        
+        // Task
+        let session = NSURLSession.sharedSession()
+        let task = session.uploadTaskWithRequest(request, fromData: requestBody) { (data, response, error) in
+            completion?(error: error)
+        }
+        
+        task.resume()
+    }
+    
+    
+    
+    // MARK: - Private Helpers
+    
+    /// Returns a new (randomized) Boundary String
+    ///
+    private func boundaryForRequest() -> String {
+        return "Boundary-" + NSUUID().UUIDString
+    }
+    
+    
+    /// Returns the Body for a Gravatar Upload OP.
+    ///
+    /// - Parameters:
+    ///     - gravatarData: The NSData-Encoded Image
+    ///     - account: The account that will get updated
+    ///     - boundary: The request's Boundary String
+    ///
+    /// - Returns: A NSData instance, containing the Request's Payload.
+    ///
+    private func bodyWithGravatarData(gravatarData: NSData, account: String, boundary: String) -> NSData {
+        let body = NSMutableData()
+        
+        // Image Payload
+        body.appendString("--\(boundary)\r\n")
+        body.appendString("Content-Disposition: form-data; name=\(UploadParameters.imageKey); ")
+        body.appendString("filename=\(UploadParameters.filename)\r\n")
+        body.appendString("Content-Type: \(UploadParameters.contentType);\r\n\r\n")
+        body.appendData(gravatarData)
+        body.appendString("\r\n")
+        
+        // Account Payload
+        body.appendString("--\(boundary)\r\n")
+        body.appendString("Content-Disposition: form-data; name=\"\(UploadParameters.accountKey)\"\r\n\r\n")
+        body.appendString("\(account)\r\n")
+        
+        // EOF!
+        body.appendString("--\(boundary)--\r\n")
+        
+        return body
+    }
+    
+    
+    
+    // MARK: - Private Properties
+    private let accountEmail    : String
+    private let accountToken    : String
+    
+    // MARK: - Private Structs
+    private struct UploadParameters {
+        static let endpointURL          = "https://api.gravatar.com/v1/upload-image"
+        static let HTTPMethod           = "POST"
+        static let compressionRatio     = CGFloat(0.9)
```

Comment:
I'm not sure the remote service is the best place to specify the compression ratio.  If a jpg format is preferred I think the remote should use a ratio of 1.0 and let the caller worry about any preferred image optimizations. Tho since a gravatar can be resized its probably best to always upload at best quality.  Maybe a png format would be best after all?